### DISCUSSION TITLE
Voxel correspondance between Non-Linearly aligned Volumes

### DISCUSSION THREAD

**Discussion by jrdimale** [Project-MONAI/MONAI#7061] [Q&A]
Greetings!

I am working on my PhD using 3D Brain MRI Data. I'm using Dipy to perform linear and non-linear registration between my volumes and a template brain (colin_27).

For each of my volumes, I first perform affine registration, performing 3D translations and rigid transformations.

Afterwards, I use the linearly aligned volumes and transform them using a deformation field, obtaining deformed 3D brain MRIs that are registered to the atlas perfectly.

As a result, I have a group of linearly aligned volumes, and another group of non-linearly aligned volumes, all referred to the same 3D brain MRI Template.

To perform a voxel-wise comparison between my volumes, I would like, for each voxel on the template, to obtain the corresponding voxel of each of my linearly aligned volumes. I believe this correspondence can be obtained by using the deformation field applied.

Thus, my objective is for each voxel (x, y, z) to obtain the corresponding voxel (x + dx, y + dy, z + dz ) for each linearly aligned volume.

For now I'm working with Dipy and have not been successful yet. I would like to know if MONAI provides a solution for my problem.

Thanks in advance for any solutions.

**Comment by KumoLiu** [comment]
Hi @jrdimale, perhaps you could check these tutorials.
https://github.com/Project-MONAI/tutorials/tree/main/3d_registration
Thanks!
